pkg/apis/serving/v1alpha1/revision_lifecycle.go

```diff
@@ -211,6 +212,31 @@ func (rs *RevisionStatus) MarkContainerMissing(message string) {
 	revCondSet.Manage(rs).MarkFalse(RevisionConditionContainerHealthy, "ContainerMissing", message)
 }
 
+// PropagateAutoscalerStatus propagates autoscaler's status to the revision's status.
+func (rs *RevisionStatus) PropagateAutoscalerStatus(ps *av1alpha1.PodAutoscalerStatus) {
+	// Propagate the service name from the PA.
+	rs.ServiceName = ps.ServiceName
+
+	// Reflect the PA status in our own.
+	cond := ps.GetCondition(av1alpha1.PodAutoscalerConditionReady)
```

Comment:
can it be folded into switch `switch a:=b {`?

Reply:
Can't really  because we're checking nil as well as cond.Status, but I moved things around.